apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-tabs.tsx

```diff
@@ -24,33 +25,36 @@ import { useWorkflowEditorContext } from '../../hooks';
 import { flattenIssues } from '../../step-utils';
 import { CustomStepControls } from '../controls/custom-step-controls';
 import { useStep } from '../use-step';
+import { useStepEditorContext } from '@/components/workflow-editor/steps/hooks';
 
 const tabsContentClassName = 'h-full w-full px-3 py-3.5 overflow-y-auto';
 
 export const InAppTabs = ({ workflow, step }: { workflow: WorkflowResponseDto; step: StepDataDto }) => {
+  const navigate = useNavigate();
   const { stepSlug = '', workflowSlug = '' } = useParams<{ workflowSlug: string; stepSlug: string }>();
   const { resetWorkflowForm } = useWorkflowEditorContext();
+  const { refetch: refetchStep } = useStepEditorContext();
   const { step: workflowStep } = useStep();
-  const { dataSchema, uiSchema } = step.controls;
-  const navigate = useNavigate();
-  const schema = buildDynamicZodSchema(dataSchema ?? {});
+
+  const { dataSchema, uiSchema, values } = step.controls;
+  const schema = useMemo(() => buildDynamicZodSchema(dataSchema ?? {}), [dataSchema]);
+  const newFormValues = useMemo(() => merge(buildDefaultValues(uiSchema ?? {}), values), [uiSchema, values]);
```

Comment:
merge the default values with the control values to set the form values initial state